R/interface.R

```diff
@@ -617,3 +621,15 @@ is_uncalled_generator <- function(sys) {
     length(code) == 2 &&
     rlang::is_call(code[[2]], "dust_system")
 }
+
+
+## This will move into mcstate eventually, but this will let us pack
+## and unpack state, once we work out what we need to do with it.
+## There are a few options here such as:
+##
+## * give me the indices of logical compartments 'a', 'b' and 'c'
+## * convert this vector/array into a list of vectors/arrays
+packer <- function(packing_state) {
+  i <- lengths(packing_state) == 0
+  mcstate2::mcstate_packer(names(packing_state)[i], packing_state[!i])
```

Comment:
```suggestion
  mcstate2::mcstate_packer(array = packing_state)
```

once https://github.com/mrc-ide/mcstate2/pull/62 is merged, though more likely we'll do


```r
  res$packer_state <- mcstate2::mcstate_packer(array = res$packing_state)
  if (!is.null(res$packing_gradient)) {
    res$packer_gradient <- mcstate2::packer(array = res$packing_gradient)
  }
```

at the top of the changes in interface.R and then drop this wrapper